src/components/modals/FlipInputModal.js

```diff
@@ -85,11 +88,16 @@ class FlipInputModalComponent extends React.PureComponent<Props, State> {
     super(props)
     this.state = {
       overridePrimaryExchangeAmount: props.overrideExchangeAmount ?? props.overridePrimaryExchangeAmount,
-      forceUpdateGuiCounter: 0
+      forceUpdateGuiCounter: 0,
+      nativeAmount: '',
+      exchangeAmount: ''
     }
   }
 
-  handleCloseModal = () => this.props.bridge.resolve()
+  handleCloseModal = () => {
+    const { nativeAmount, exchangeAmount } = this.state
+    this.props.bridge.resolve(Promise.resolve({ nativeAmount, exchangeAmount }))
```

Comment:
Can this modal close when the nativeAmount and/or exchangeAmount are still equal to empty string?
And if so, what would happen with whoever calls this modal? Can they handle the empty string?